site/content/agent-app-widgets/js-api.md

```diff
@@ -101,7 +101,191 @@ It appends given message at the end of current conversation input window or into
+
+### How to update a section
+
+To update a section you can use the `modifyCustomerDetailsSection` method like this:
+
+```js
+LiveChat.modifyCustomerDetailsSection(
+  { 
+    "title": "Example section",
+    "components": [
+      {
+        "type": "button",
+        "data": {
+          "label": "Updated button",
+          "id": "example-button"
+        }
+      }
+    ]
+  } 
+);
+```
+---
+**IMPORTANT**
+
+The **title** attribute in the root of the JSON acts as an ID to the section and *cannot* be changed. You need to provide it to identify which section you want to update.
+
+---
+
+### Component types
+Components are *lego bricks* which can be used for building a section.
+
+#### Title
+Title could be used in several cases. Component look depends on given data.
+
+| Property | Required | Type | Note |
+|----------|----------| ---- | ----------- |
+| `title` | Yes | string |                   |
+| `value` | No | string |                    |
+| `description` | No | string |              |
+| `imgUrl` | No | string |                   |
+| `imgSize` | No | string, one of: ["small", "big"]` | Default value: "big" |
+| `link` | No | string | URL added when title is a link. |
+
+> Example of title component
+
+```json
+{
+    type: 'title',
+    data: {
+        title: 'title',
+        value: 'value',
+        description: 'description',
+        imgUrl: 'https://www.gstatic.com/webp/gallery/4.jpg'
+    }
+}
+```
+
+#### Button
+Simple button from design system
+
+| Property | Required | Type   |
+|----------|----------| ------ |
+| `title`  | Yes      | string |
+| `value`  | Yes      | string | 
+
+> Example of button component
+
+```json
+{
+    type: 'button',
+    data: {
+          label: 'second button',
+          id: 'second-button'
+    }
+}
+```
+
+#### Label with value
+
+| Property | Required | Type    | Description |
+|----------|----------| ------- | ----------- |
+| `label`  | No       | string  |             |
+| `value`  | No       | string  |             |
+| `inline` | No       | boolean | Default value: `true` |
+
+> Example of label with value component
+
+```json
+{
+  type: 'label_value',
+  data: {
+    label: 'Name',
+    value: 'Stefan',
+  }
+}
+```
+
+#### Link
+
+| Property | Required | Type    | Description |
+|----------|----------| ------- | ----------- |
+| `url`    | Yes      | string  |             |
+| `value`  | No       | string  |             |
+| `target` | No       | string  | Same as HTML attribute |
+| `inline` | No       | boolean |             |
+
+> Example of link component
+
+```json
+{
+  type: 'link',
+  data: {
+    value: 'click me',
+    url: 'google.com'
+  }
+}
+```
+
+#### Line
+Line could be used to separate section content. It has no components inside. 
+
+> Example of line component
+
+```json
+{
+    type: 'line'
+}
+```
+
+#### Section
```

Comment:
We already discussed this with @TomaszRatajczyk some time ago and come to a conclusion that this part should come first before any other component explanation.